### PR TITLE
Sscs 5094 send direction notice via letter

### PR DESCRIPTION
**Pull request by arcotc**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-5094

### Change description ###
Send Direction Notice via letter

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ private void sendLetterNotification(NotificationWrapper wrapper, Subscription su
     }
 
     private void sendMandatoryLetterNotification(NotificationWrapper wrapper, Notification notification, SubscriptionType subscriptionType) {
-        if (hasLetterTemplate(notification) && isMandatoryLetterEventType(wrapper)) {
+        if (isMandatoryLetterEventType(wrapper)) {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: This will cause a NPE if there is no letter template configured for the event type in the application.yaml.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 2]
```diff
@@ -142,14 +142,14 @@ private void sendMandatoryLetterNotification(NotificationWrapper wrapper, Notifi
 
             if (bundledLettersOn && isBundledLetter(wrapper.getNotificationType())) {
                 sendBundledLetterNotificationToAppellant(wrapper, notification, subscriptionType);
-            } else {
+            } else if (hasLetterTemplate(notification)) {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: No need to have this check if you put it back on line 136.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 3]
```diff
@@ -142,14 +142,14 @@ private void sendMandatoryLetterNotification(NotificationWrapper wrapper, Notifi
...
                 notificationHandler.sendNotification(wrapper, notification.getLetterTemplate(), NOTIFICATION_TYPE_LETTER, sendNotification);
             }
         }
     }
 
     private void sendFallbackLetterNotification(NotificationWrapper wrapper, Subscription subscription, Notification notification, SubscriptionWithType subscriptionWithType, NotificationEventType eventType) {
-        if (hasNoSubscriptions(subscription) && hasLetterTemplate(notification) && isFallbackLetterRequired(wrapper, subscriptionWithType, subscription, eventType, notificationValidService)) {
+        if (hasNoSubscriptions(subscription) && isFallbackLetterRequired(wrapper, subscriptionWithType, subscription, eventType, notificationValidService)) {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: Same comment as above (136)

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 4]
```diff
@@ -158,7 +158,7 @@ private void sendFallbackLetterNotification(NotificationWrapper wrapper, Subscri
 
             if (bundledLettersOn && isBundledLetter(wrapper.getNotificationType())) {
                 sendBundledLetterNotificationToAppellant(wrapper, notification, subscriptionWithType.getSubscriptionType());
-            } else {
+            } else if (hasLetterTemplate(notification)) {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: Same comment as above on (152)

File: src/test/java/uk/gov/hmcts/reform/sscs/service/NotificationServiceTest.java [hunk 5]
```diff
@@ -61,7 +61,7 @@
     private static final String CASE_ID = "1000001";
     private static final String EMAIL_TEMPLATE_ID = "email-template-id";
     private static final String SMS_TEMPLATE_ID = "sms-template-id";
-    private static final String LETTER_TEMPLATE_ID_STRUCKOUT = "struckOut";
+    private static final String LETTER_TEMPLATE_ID_STRUCKOUT = null;
```
Comment (arcotc) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/NotificationServiceTest.java]: Is this required?  If it is, I find it a little strange that it is set to `null`.  I don't think there are templateIds for bundled letters any more.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 6]
```diff
@@ -306,4 +306,8 @@ protected String getBundledLetterTemplate(NotificationEventType notificationEven
         }
         return bundledLetterTemplate;
     }
+
+    private boolean hasBundledLetterTemplate(NotificationWrapper wrapper) {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: We don't need this method.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 7]
```diff
@@ -215,7 +215,7 @@ private static boolean isValidLetterAddress(Address addressToUse) {
     }
 
     private void sendBundledLetterNotificationToAppellant(NotificationWrapper wrapper, Notification notification, SubscriptionType subscriptionType) {
-        if (hasLetterTemplate(notification)) {
+        if (hasBundledLetterTemplate(wrapper)) {
```
Comment (arcotc) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: We don't need this check, as bundled letters don't have templates.  They exist in sscs-pdf-email-common

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java [hunk 8]
```diff
@@ -18,6 +18,7 @@
 public class NotificationValidService {
     private static final List<NotificationEventType> MANDATORY_LETTER_EVENT_TYPES = Arrays.asList(STRUCK_OUT, HEARING_BOOKED_NOTIFICATION);
```
Comment (oleg-hmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java]: Just a small readability suggestion - keep names in alphabetical order. It is not an issue here as the lists are small.

File: src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java [hunk 9]
```diff
@@ -321,6 +330,30 @@ public void getRepSalutationWhenRepHasOrgAndName() {
...
+    @Test
+    @Parameters(method = "nonBundledLetterTemplates")
+    public void invalidBundledLetterTemplate(NotificationEventType eventType) {
+        assertNull(classUnderTest.getBundledLetterTemplate(eventType, buildBaseWrapper(APPELLANT_WITH_ADDRESS, eventType).getNewSscsCaseData()));
+    }
+
+    @Test
+    @Parameters(method = "bundledLetterTemplates")
+    public void validBundledLetterType(NotificationEventType eventType) {
+        assertNotNull(getBundledLetterFileType(eventType, buildBaseWrapper(APPELLANT_WITH_ADDRESS, eventType).getNewSscsCaseData()));
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java]: Add tests for the value to equal STRIKE_OUT_NOTICE, DIRECTION_TEXT and null

File: src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java [hunk 10]
```diff
@@ -321,6 +330,30 @@ public void getRepSalutationWhenRepHasOrgAndName() {
...
+    }
+
+    @Test
+    @Parameters(method = "nonBundledLetterTemplates")
+    public void invalidBundledLetterTileType(NotificationEventType eventType) {
+        assertNull(getBundledLetterFileType(eventType, buildBaseWrapper(APPELLANT_WITH_ADDRESS, eventType).getNewSscsCaseData()));
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/SendNotificationServiceTest.java]: Add tests for the value to equal STRIKE_OUT_NOTICE, DIRECTION_TEXT and null
